test/unstable/ref-maybe-const-forall-intent.good

```diff
@@ -7,7 +7,7 @@ ref-maybe-const-forall-intent.chpl:87: warning: inferring a 'ref' intent on an a
 ref-maybe-const-forall-intent.chpl:89: warning: inferring a 'ref' intent on an array in a forall is unstable - in the future this may require an explicit 'ref' forall intent for 'myArray1'
 ref-maybe-const-forall-intent.chpl:92: warning: inferring a 'ref' intent on an array in a forall is unstable - in the future this may require an explicit 'ref' forall intent for 'myArray1'
 ref-maybe-const-forall-intent.chpl:98: In method 'foo':
-ref-maybe-const-forall-intent.chpl:99: warning: inferring a 'ref' intent on an array in a forall is unstable - in the future this may require an explicit 'ref' forall intent for 'A'
+ref-maybe-const-forall-intent.chpl:99: warning: inferring a 'ref' intent on an array in a forall is unstable - in the future this may require an explicit 'ref' forall intent for 'this'
```

Comment:
I'm trying to understand this. So, for methods, we also require an explicit intent for `this`. Though, it looks awkward to ask for a `with (ref this)` for a loop that doesn't refer to `this`. (I am assuming https://github.com/chapel-lang/chapel/pull/23612/files#diff-824d818539798ec50ff6674251a6f16ad6189714b435b5f249469955238af094R100 is the line where this warning fires).

So, would adding `with (ref this)` be the fix? What would happen if I did `with (ref A)` instead?

Reply:
Yes, in this case you have to do `with (ref this)`.

If you do `with (ref A)`, you get `error: cannot reference a field or function 'A' in a with-clause of this forall loop`. This directs you to using `with (ref this)`, so by trying to use the correct name in the warning it hopefully makes for a nicer transition for users.

Reply:
I see. This is nothing to hold the PR for, but I think this requires some more discussion. On a quick look I am not seeing any issue that captures my concern. Do you know any offhand? Otherwise, I can create a new one.